source-code-git/fs/src/interface.ts

```diff
@@ -8,17 +8,18 @@
  */
 export type NodeishFilesystem = {
 	writeFile: (path: string, data: string) => Promise<void>
-	readFile: (path: string, options: { encoding: "utf-8" }) => Promise<FileData>
+	/* NOTE: accept options as 'any' for now until we figure out how to get the node types to match */
+	readFile: (path: string, options?: any) => Promise<FileData>
 	readdir: (path: string) => Promise<string[]>
 	/**
 	 * https://nodejs.org/api/fs.html#fspromisesmkdirpath-options
 	 *
 	 * Upon success, fulfills with undefined if recursive is false, or the first directory path created if recursive is true.
 	 */
-	mkdir:
-		| ((path: string) => Promise<void>)
-		| ((path: string, options?: { recursive: boolean }) => Promise<string>)
+	mkdir: (path: string, options?: { recursive: boolean }) => Promise<string | undefined>
 	rm: (path: string, options?: { recursive: boolean }) => Promise<void>
+	rmdir: (path: string, options?: any) => Promise<void>
```

Comment:
`rmdir` has the same issue. If we allow `any` for the options, developers will use the filesystem differently and since we do not check for options that are being passed, different filesystem implementations will yield different results.